### PR TITLE
fix(search): L3-4161 Search Close On Result Selection

### PR DESCRIPTION
**Pull request by adietrich3074**
**Jira ticket**

[L3-4161](https://phillipsauctions.atlassian.net/browse/L3-4161)

**Summary**

Added handling of closing search upon user clicking a result

**Change List (describe the changes made to the files)**

Passed showSearch function into Search Result component to allow for search result selection to trigger search close

**Acceptance Test (how to verify the PR)**

Click a search result and then go back to the page that you searched from and the search should be closed 

**Things to look for during review**

- [ ] PR title should correctly describe the most significant type of commit. I.e. `feat(scope): ...` if a `minor` release should be triggered.
- [ ] All commit messages follow convention and are appropriate for the changes
- [ ] All references to `phillips` class prefix are using the prefix variable
- [ ] All major areas have a `data-testid` attribute.
- [ ] Document all props with jsdoc comments
- [ ] All strings should be translatable.
- [ ] Unit tests should be written and should have a coverage of 90% or higher in all areas.

New Components

- [ ] Are there any [accessibility considerations](https://www.w3.org/WAI/ARIA/apg/patterns/) that need to be taken into account and tested?
- [ ] Default story called "Playground" should be created for all new components
- [ ] Create a jsdoc comment that has an Overview section and a link to the Figma design for the component
- [ ] Export the component and its typescript type from the `index.ts` file
- [ ] Import the component scss file into the `componentStyles.scss` file.

[L3-4161]: https://phillipsauctions.atlassian.net/browse/L3-4161?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/Search/SearchResults/SearchResults.tsx [hunk 1]
```diff
@@ -40,11 +41,14 @@ const SearchResults = ({
       <ul data-testid="search-results" className={`${px}-search__results-container`}>
         {isLoading ? <li className={`${px}-search__result`}>{loadingText}</li> : null}
         {hasResults &&
-          autoCompleteResults.map((result) => {
+          autoCompleteResults.map((result, i) => {
+            i++;
```
Comment (sgrund14) [inline src/components/Search/SearchResults/SearchResults.tsx:45]: why is this incrementing necessary?

Comment (adietrich3074) [reply src/components/Search/SearchResults/SearchResults.tsx:45]: I guess it really isn't but then the testId of the first result is search-result-0 😆

File: src/components/Search/SearchResults/SearchResults.tsx [hunk 2]
```diff
@@ -40,11 +41,13 @@ const SearchResults = ({
       <ul data-testid="search-results" className={`${px}-search__results-container`}>
         {isLoading ? <li className={`${px}-search__result`}>{loadingText}</li> : null}
         {hasResults &&
-          autoCompleteResults.map((result) => {
+          autoCompleteResults.map((result, i) => {
             return (
-              <li key={result.id} className={`${px}-search__result`}>
+              <li key={result.id} className={`${px}-search__result`} onClick={() => showSearchResults(false)}>
```
Comment (sgrund14) [inline src/components/Search/SearchResults/SearchResults.tsx:46]: might be good to put this onClick on the `Link` component? what you think @scottdickerson

File: src/components/Search/SearchResults/SearchResults.tsx [hunk 3]
```diff
@@ -16,6 +16,7 @@ export interface SearchResultsProps extends React.HTMLAttributes<HTMLElement> {
   loadingText?: string;
   onKeyDown?: (event: React.KeyboardEvent<HTMLAnchorElement>) => void;
   userInputValue?: string;
+  closeSearch: (value: boolean) => void;
```
Comment (sgrund14) [inline src/components/Search/SearchResults/SearchResults.tsx:19]: naming slightly confusing here, if it's accepting the boolean as a prop. could just keep it as `showSearch`
or, where you pass it in `Search.tsx`, do this:

```
closeSearch={() => showSearch(false)}
```

**Review by sgrund14 — COMMENTED** [review]
just that comment about prop naming, but otherwise LGTM

**Review by sgrund14 — APPROVED** [review]

**Review by constantinehuzenko — APPROVED** [review]
LGTM

**Pull request merged**